frontend/src/metabase/visualizations/click-actions/actions/HideColumnAction/HideColumnAction.tsx

```diff
@@ -48,9 +49,18 @@ export const HideColumnAction: LegacyDrill = ({
           };
         }
 
-        return onUpdateVisualizationSettings({
-          "table.columns": columnSettingsCopy,
-        });
+        const vizSettings = { "table.columns": columnSettingsCopy };
+
+        const onUpdateVisualizationSettingsSdk = clicked?.extraData
+          ?.onUpdateVisualizationSettingsSdk as (
+          settings: VisualizationSettings,
+        ) => void;
+
+        // The embedding sdk does not store the question in the query builder store,
+        // therefore we must update the visualization settings in the React context.
+        onUpdateVisualizationSettingsSdk?.(vizSettings);
```

Comment:
If `onUpdateVisualizationSettingsSdk` is passed into `extraData` which only happens we are in a SDK context, we also call this handler to update the SDK's question via the `useLoadQuestion` React context